training_utils.py

```diff
@@ -377,4 +431,69 @@ def train_loop(
         print(f"Saved final model to {save_path}")
 
     print("Best Perplexities,", best_ppl)
-    return
\ No newline at end of file
+    return
+
+def reg_loop(
+    args,
+    model,
+    train_dataset,
+    device,
+    tokenizer=None,
+    regularizer=None
+):
+    # Quick hack to get tree regularization scores for a saved model
```

Comment:
Does this comment apply to the entire `reg_loop`? The functionality of this is unclear, a docstring would be helpful~

Reply:
Yup, it applies to the entire loop. Added docstring.

Reply:
Thanks!